array-manipulation-1/src/compact.js

```diff
@@ -0,0 +1,17 @@
+/* exported compact */
+function compact(array) {
+  const result = [];
+  for (const i of array) {
+    if (
+      i !== false &&
+      i !== null &&
+      Number.isNaN(i) === false &&
+      i !== 0 &&
+      i !== undefined &&
+      i !== ''
+    ) {
+      result.push(i);
+    }
+  }
+  return result;
+}
```

Comment:
We can simplify your conditional a lot. As we were talking about, we only need to check if the value is truthy or not. Since a conditional expression will always evaluate to either `true` or `false`, it's quite literally all we need to check!

```
function compact(array) {
  const output = [];
  for (let i = 0; i < array.length; i++) {
    const element = array[i];
    if (element) { 
      output.push(element);
    }
  }
  return output;
}
```

Reply:
its so beautiful :')